File: frontend/src/types/execution.ts
// types/execution.ts
// 执行监控类型：与后端 /api/v1/executions 对齐（execution.md：列表为 execution_id 分组聚合，详情含链路）
export type ExecutionStepType = 'llm' | 'rag' | 'tool'
export type ExecutionStatus = 'success' | 'error'

export interface ExecutionStep {
  id: number
  step_type: ExecutionStepType
  step_name: string
  status: ExecutionStatus
  input_json: Record<string, unknown> | null
  output_json: Record<string, unknown> | null
  duration_ms: number
  created_at: string
}

export interface LLMUsage {
  id: number
  provider: string
  model: string
  round: number
  input_tokens: number
  output_tokens: number
  total_tokens: number
  latency_ms: number
  created_at: string
}

export interface ExecutionSummary {
  execution_id: string
  agent_id: number
  agent_name: string | null
  conversation_id: number
  user_id: number
  started_at: string
  finished_at: string
  duration_ms: number
  step_count: number
  error_steps: number
  status: ExecutionStatus
  total_tokens: number
  llm_latency_ms: number
}

export interface ExecutionDetail extends ExecutionSummary {
  steps: ExecutionStep[]
  usages: LLMUsage[]
}

export interface ExecutionListResponse {
  items: ExecutionSummary[]
  total: number
}

export interface ExecutionListParams {
  agent_id?: number
  conversation_id?: number
  status?: ExecutionStatus
  limit?: number
  offset?: number
}

File: backend/app/schemas/execution.py
# schemas/execution.py
# 执行监控响应模型（需求 3.8 / 4.13 / 4.14，execution.md）：列表为 execution_id 分组聚合，详情含步骤链路与用量
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class ExecutionStepItem(BaseModel):
    """执行步骤（4.13）：LLM 轮次 / RAG 检索 / 工具调用"""

    id: int
    step_type: str
    step_name: str
    status: str
    input_json: dict[str, Any] | None
    output_json: dict[str, Any] | None
    duration_ms: int
    created_at: datetime


class LLMUsageItem(BaseModel):
    """LLM 调用用量（4.14）：tool-calling 多轮产生多条"""

    id: int
    provider: str
    model: str
    round: int
    input_tokens: int
    output_tokens: int
    total_tokens: int
    latency_ms: int
    created_at: datetime


class ExecutionSummary(BaseModel):
    """execution 聚合摘要（列表项）：总耗时为各步骤 duration_ms 合计（毫秒精确）"""

    execution_id: str
    agent_id: int
    agent_name: str | None
    conversation_id: int
    user_id: int
    started_at: datetime
    finished_at: datetime
    duration_ms: int
    step_count: int
    error_steps: int
    status: str
    total_tokens: int
    llm_latency_ms: int


class ExecutionListResponse(BaseModel):
    """execution 分页列表（D7：聚合查询带总数）"""

    items: list[ExecutionSummary]
    total: int


class ExecutionDetail(ExecutionSummary):
    """执行详情：汇总 + 步骤链路 + LLM 用量"""

    steps: list[ExecutionStepItem]
    usages: list[LLMUsageItem]
